__tests__/commands/publish.js

```diff
@@ -0,0 +1,71 @@
+/* @flow */
+
+import {run as buildRun} from './_helpers.js';
+import {run as publish} from '../../src/cli/commands/publish.js';
+import {ConsoleReporter} from '../../src/reporters/index.js';
+
+const path = require('path');
+
+const fixturesLoc = path.join(__dirname, '..', 'fixtures', 'publish');
+
+const setupMocks = function(config) {
+  // Mock actual network request so that no package is actually published.
+  // $FlowFixMe
+  config.registries.npm.request = jest.fn();
+  config.registries.npm.request.mockReturnValue(
+    new Promise(resolve => {
+      resolve({status: 200});
+    }),
+  );
+
+  // Mock the npm login name. Otherwise yarn will prompt for it and break CI.
+  // $FlowFixMe
+  config.registries.npm.getAuth = jest.fn();
+  config.registries.npm.getAuth.mockReturnValue('test');
+};
+
+const runPublish = buildRun.bind(
+  null,
+  ConsoleReporter,
+  fixturesLoc,
+  async (args, flags, config, reporter, lockfile, getStdout): Promise<string> => {
+    setupMocks(config);
+    await publish(config, reporter, flags, args);
+    return getStdout();
+  },
+);
+
+test.concurrent('publish should default access to undefined', () => {
+  return runPublish([], {newVersion: '0.0.1'}, 'minimal', config => {
+    const requestCallParams = config.registries.npm.request.mock.calls[0][1];
```

Comment:
I find it more readable to use `toBeCalledWith` rather than reaching into the mock objects

Reply:
I wasn't sure if there was a way to do that and only specify the parameter that I care about.

Reply:
or rather, the 2nd parameter is an object with a bunch of properties but I'm realy only testing 1 of them.

Reply:
It seems I had to nest a couple `expect.objectContaining()` calls, but I cleaned it up (or made it longer and more nested, depending on your point of view 😄 )